### DISCUSSION TITLE
can windows port log to console?

### DISCUSSION THREAD

**Discussion by garbb** [FreeRDP/FreeRDP#8375] [Q&A] [answered]
For wfreerdp.exe in windows, I am able to log to a file in the same directory as wfreerdp.exe by setting the following environment variables:
```
SET WLOG_LEVEL=DEBUG
SET WLOG_APPENDER=file
SET WLOG_FILEAPPENDER_OUTPUT_FILE_NAME=wfreerdp.log
SET WLOG_FILEAPPENDER_OUTPUT_FILE_PATH=.
```

Is there any way to have wfreerdp.exe log to the console stdout?
I found [this](https://github.com/FreeRDP/FreeRDP/blob/master/docs/wlog.md) which says that you can set WLOG_APPENDER=console and it should log to stdout, but this does not seem to do anything when wfreerdp.exe is run in a console window. Is this just not implemented for the windows port?

**Comment by akallabeth** [comment, marked as answer]
This is a limitation of windows applications.
you can build with `-DWITH_WIN_CONSOLE=ON` and have console output, but you will always have a console window open beside `wfreerdp` then.
